introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs

```diff
@@ -1,19 +1,74 @@
 use crate::SqlIntrospectionResult;
 use datamodel::{
-    common::{PrismaType, PrismaValue},
+    common::{names::NameNormalizer, PrismaType, PrismaValue},
     dml, Datamodel, Field, FieldArity, FieldType, IdInfo, IdStrategy, Model, OnDeleteStrategy, RelationInfo,
     ScalarListStrategy,
 };
 use log::debug;
+use prisma_inflector;
 use regex::Regex;
 use sql_schema_describer::*;
 
+ fn is_migration_table(table: &Table) -> bool {
+    table.name == "_Migration"
+}
+
+ fn is_many_to_many_relation_table(table: &Table) -> bool {
+    table.name.starts_with("_")
+        && table.columns.iter().count() == 2
+        && table.foreign_keys.iter().count() == 2
+        && table.columns.iter().find(|column| column.name == "A").is_some()
+        && table.columns.iter().find(|column| column.name == "B").is_some()
+}
+
+ fn is_scalar_list_table(table: &Table) -> bool {
+    table.name.contains("_")
+        && table.columns.iter().count() == 3
+        && table.columns.iter().find(|column| column.name == "nodeId").is_some()
+        && table.columns.iter().find(|column| column.name == "position").is_some()
+        && table.columns.iter().find(|column| column.name == "value").is_some()
+}
+
+    fn create_many_to_many_field(foreign_key: &ForeignKey, relation_table_name: &str) -> Field {
```

Comment:
Identation.

It's a good idea to install [rustfmt](https://github.com/rust-lang/rustfmt) for style fixes. Most of the editors allow you to run it automatically on every save. Please use it :)

Another useful tool is [clippy](https://github.com/rust-lang/rust-clippy). Run it every now and then to the code to get very useful tips for possible problems.